### PR TITLE
Try catch repair

### PR DESCRIPTION
**Pull request by danglotb**
I proposed the following changes: 
* add a way to specify a line to process the repair, or not according to the fourth strategy. 

Changes in instrumentation results with this:
```java
} catch (ForceReturn _bcornu_return_t) {
    if ((-1 == (_bcornu_return_t.getTargetLine())) || (6 == (_bcornu_return_t.getTargetLine())))
        return ((String) (_bcornu_return_t.getDecision().getValue()));
    else
        throw _bcornu_return_t;    
} finally {
    _bcornu_methode_context1.methodEnd();
}
```

### CODE REVIEW

**Comment by danglotb** [comment]
Hello, @bcornu sorry, i miss clicked.

**Comment by tdurieux** [comment]
Done?

File: src/main/java/fr/inria/spirals/npefix/main/all/TryCatchRepairStrategy.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Created by Benjamin DANGLOT
+ * benjamin.danglot@inria.fr
+ * on 11/07/17
+ */
+@SuppressWarnings("all")
+public class TryCatchRepairStrategy implements RepairStrategy {
```
Comment (tdurieux) [inline src/main/java/fr/inria/spirals/npefix/main/all/TryCatchRepairStrategy.java]: Extends defaultStrategy?

File: src/main/java/fr/inria/spirals/npefix/transformer/processors/TryCatchRepair.java [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Created by Benjamin DANGLOT
+ * benjamin.danglot@inria.fr
+ * on 11/07/17
+ */
+@SuppressWarnings("all")
+public class TryCatchRepair extends AbstractProcessor<CtMethod> {
```
Comment (danglotb) [inline src/main/java/fr/inria/spirals/npefix/transformer/processors/TryCatchRepair.java]: Extends MethodEncapsulation. A lot of code has been duplicated.

**Pull request merged**